src/util/bip143.rs

```diff
@@ -120,6 +231,18 @@ mod tests {
         witness_script
     }
 
+    fn run_test_sighash_bip143(tx: &str, script: &str, input_index: usize, value: u64, hash_type: u32, expected_result: &str) {
+        let tx: Transaction = deserialize(&Vec::<u8>::from_hex(tx).unwrap()[..]).unwrap();
+        let script = Script::from(Vec::<u8>::from_hex(script).unwrap());
+        let mut raw_expected = Vec::<u8>::from_hex(expected_result).unwrap();
```

Comment:
You should be able to directly do `SigHash::from_hex` (unless there's yet another reversing nuance shit here that I don't remember)

Reply:
done